Automattic/newspack-plugin
# Checklist Component

### All Submissions:






### Changes proposed in this Pull Request:

Introduces a checklist component, based somewhat on https://github.com/Automattic/wp-calypso/tree/master/client/components/checklist. The component displays a list of tasks, each with a `title`, `description`, `buttonText`, and `completedTitle` which is displayed when the task has been committed. Each `Task` has `onClick` and `onSkip` events which fire when the primary or "Skip" buttons are clicked. The `Checklist` takes a `currentTask` integer as a prop which determines how far along in the list the user has gotten. In addition to visual differences between upcoming/active/complete `Task` there is a `ProgressBar` element that visualizes the user's progress. 

This is the mockup the Component was built to match:

<img width="542" alt="Screen Shot 2019-05-14 at 7 44 00 PM" src="https://user-images.githubusercontent.com/1477002/57739236-b79e3100-7680-11e9-9d53-083c615c9947.png">

Some questions for discussion:

1) In the example implementation, the `onClick` and `onSkip` events both will advance the `currentTask`. This generally works well unless the user clicks `Edit` to make a previously completed `Task` current. At this point things get a little weird because completing that task again takes you to the next one, which may be a step backwards overall. @claudiulodro I'm curious to hear if you think this is the right model for handling Click/Skip/Edit buttons?

2) The Primary buttons have less horizontal padding than the mockups show. Rather than tinker with their styling in this PR, curious if we should make changes to the `Button` component to more closely match the mocks in this context. cc: @philipjohn 

3) The mockups also introduce a light text-only button (Edit/Skip). For this I used the `isLink` style button, and added a new style definition to the `Button` component. @philipjohn curious what you think of this, if `isLink` should be held for a different use case, and if so how to handle this new button style.

4) The mockups show an "I'm done setting up" button at the bottom. Is this meant to be part of this component, or is it a free-floating button?

Closes #31, #45 

### How to test the changes in this Pull Request:

1. `npm ci && npm run clean && npm run build:webpack`
2. Navigate to `/admin.php?page=newspack-subscriptions-wizard`
3. Observe and interact with the component, trying Primary, Edit and Skip buttons. After some tasks have been completed there should be Hide/Show Completed UI in the upper right corner. Verify that this is working as expected.

### Other information:
